### ISSUE TITLE
[Feature] Validation Loss and data split for SFT

### ISSUE THREAD

**Issue by weicheng59** [InternLM/InternLM#164]
### Describe the feature

It is inconvenient when user try to determine the best timing to stop training. 
Validation loss that calculate every a few steps during the training process will greatly help user to understand if the model is overfitting or underfitting. 
Right now we can save serval checkpoints to test the validation loss because we only have loss in log. Judge the performance of a specific task based on training loss is not reliable.

### Will you implement it?

- [ ] I would like to implement this feature and create a PR!

**Comment by mgyong** [comment]
how do we use this feature of the validation loss?
